### PR TITLE
Use ColPack instead of SparseDiffTools for coloration

### PR DESCRIPTION
**Pull request by tmigot**
This will simplify the dependencies of the package.

I will add SparseDiffTools as an optional dependency after this one.

Some results:
https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/blob/benchmark/benchmark/coloration.jl
https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/blob/benchmark/benchmark/coloration1000.dat

### CODE REVIEW

**Comment by tmigot** [comment]
Benchmarks:
https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/blob/benchmark/benchmark/2023-06-23_adnlpmodels_benchmark_jac_nscal_1000_mono.png

https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/blob/benchmark/benchmark/2023-06-23_adnlpmodels_benchmark_hess_nscal_1000_mono.png

File: src/ad.jl [hunk 1]
```diff
@@ -338,3 +338,35 @@ function ADModelNLSBackend(
...
+  coloring::ColPack.AbstractColoring = d1_coloring("DISTANCE_ONE"),
+  ordering::ColPack.AbstractOrdering = incidence_degree_ordering("INCIDENCE_DEGREE"),
+  )
+  return ColPackColoration{typeof(partition_choice), typeof(coloring), typeof(ordering)}(partition_choice, coloring, ordering)
+end
+
+function sparse_matrix_colors(A, alg::ColPackColoration)
+  m, n = size(A)
+  partition_by_rows = alg.partition_choice(m, n)
+  if A != spzeros(m, n)
```
Comment (amontoison) [inline src/ad.jl:361]:
```suggestion
  if !isempty(A.nzval)
```

**Comment by amontoison** [comment]
Except a minor comment, it's good for me Tangi.
Thanks!

**Review by amontoison — APPROVED** [review]

**Comment by amontoison** [comment]
@tmigot 
You removed a lot of dependencies after this PR :+1: 
```julia
(@v1.9) pkg> dev .
   Resolving package versions...
  No Changes to `~/.julia/environments/v1.9/Project.toml`
    Updating `~/.julia/environments/v1.9/Manifest.toml`
  [ec485272] - ArnoldiMethod v0.2.0
  [6a86dc24] - FiniteDiff v2.21.1
  [86223c79] - Graphs v1.8.0
  [d25df0c9] - Inflate v0.1.3
  [699a6c99] - SimpleTraits v0.9.4
  [47a9eef4] - SparseDiffTools v2.4.0
  [aedffcd0] - Static v0.8.7
  [0d7ed370] - StaticArrayInterface v1.4.0
  [19fa3120] - VertexSafeGraphs v0.2.0
  [1a1011a3] - SharedArrays
```

**Pull request merged**